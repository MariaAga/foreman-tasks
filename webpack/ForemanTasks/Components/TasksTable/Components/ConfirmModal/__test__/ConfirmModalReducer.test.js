import { testReducerSnapshotWithFixtures } from '@theforeman/test';
import {
  UPDATE_MODAL,
  CANCEL_MODAL,
  RESUME_SELECTED_MODAL,
} from '../../../TasksTableConstants';

import reducer from '../ConfirmModalReducer';

const fixtures = {
  'should return the initial state': {},
  'should handle UPDATE_MODAL to cancel': {
    action: {
      type: UPDATE_MODAL,
      payload: { modalID: CANCEL_MODAL },
    },
  },
  'should handle UPDATE_MODAL to resume': {
    action: {
      type: UPDATE_MODAL,
      payload: { modalID: RESUME_SELECTED_MODAL },
    },
  },
};

describe('ConfirmModalReducer reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
