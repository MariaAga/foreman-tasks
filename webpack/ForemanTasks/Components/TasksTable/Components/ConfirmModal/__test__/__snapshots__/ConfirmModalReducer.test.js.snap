// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModalReducer reducer should handle UPDATE_MODAL to cancel 1`] = `
Object {
  "actionState": "stopped",
  "actionText": "cancel",
  "actionType": "cancelConfirmModal",
}
`;

exports[`ConfirmModalReducer reducer should handle UPDATE_MODAL to resume 1`] = `
Object {
  "actionState": "running",
  "actionText": "resume",
  "actionType": "resumeSelectedConfirmModal",
}
`;

exports[`ConfirmModalReducer reducer should return the initial state 1`] = `Object {}`;
